### PR TITLE
[BUGFIX] Correctly fetch media for translated pages

### PR DESCRIPTION
**Pull request by cweiske**
Utilize the PageRepository to correctly fetch resources of translated pages
that are stored in pages_language_overlay.

Resolves: https://github.com/FluidTYPO3/vhs/issues/1392

### CODE REVIEW

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 1]
```diff
@@ -85,14 +92,23 @@ public function getResource($fileReference)
...
-     * @param string $table name of the table to get the file reference for
-     * @param string $field name of the field referencing a file
-     * @param integer $uid uid of the related record
+     * @param string $table  name of the table to get the file reference for
+     * @param string $field  name of the field referencing a file
+     * @param array  $record Database row
      * @return array
      */
-    protected function getFileReferences($table, $field, $uid)
+    protected function getFileReferences($table, $field, $record)
```
Comment (NamelessCoder) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: Just to protect against breaking changes for those that may have subclassed this ViewHelper: please rename this argument to `$uidOrRecord` and check/convert that as needed. If a UID is passed you can call `$this->getRecord($uidOrRecord)` to fetch the record.

Side note: we don't align the argument names/descriptions, please separate those with a single space.

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 2]
```diff
@@ -87,12 +94,27 @@ public function getResource($fileReference)
...
+    protected function getFileReferences($table, $field, $uidOrRecord)
     {
-        $fileObjects = $this->fileRepository->findByRelation($table, $field, $uid);
+        if (is_array($uidOrRecord)) {
+            $record = $uidOrRecord;
+        } else {
+            $record = $this->getRecord($uidOrRecord);
+        }
+
+        if ($table == 'pages') {
```
Comment (xf-) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: compare ===

Comment (cweiske) [reply Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: done.

**Comment by cweiske** [comment]
@xf- @NamelessCoder - I've adjusted the patch.

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php (hunk 1 above)
Comment (cweiske) [reply Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: done

**Comment by cweiske** [comment]
patch rebased.

**Pull request merged**